regulation-worker/internal/delete/batch/batch.go

```diff
@@ -31,64 +28,28 @@ import (
 
 var (
 	pkgLogger             = logger.NewLogger().Child("batch")
-	regexRequiredSuffix   = regexp.MustCompile(".json.gz$")
 	StatusTrackerFileName = "rudderDeleteTracker.txt"
-	supportedDestinations = []string{"S3"}
+	supportedDestinations = []string{"S3", "S3_DATALAKE"}
 )
 
 const listMaxItem int64 = 1000
 
-type deleteManager interface {
-	delete(ctx context.Context, patternFilePtr, targetFilePtr string) ([]byte, error)
-}
-
 type Batch struct {
 	mu         sync.Mutex
 	FM         filemanager.FileManager
```

Comment:
The thing is filehandler needs to be created on processing of every file as they store the intermediate results to allow for cleaner interface. We have added a factory to instantiate a filehandler everytime we see a file of that sorts. So this might not work.